browser/src/canvas/sections/ShapeHandlesSection.ts

```diff
@@ -617,9 +641,87 @@ class ShapeHandlesSection extends CanvasSectionObject {
 
 		(window as any).IgnorePanning = false;
 
-		// This is for tablet and mobbile but we can use this for also desktop, if we want to avoid sending mouse events for shapes to core side.
-		if (this.containerObject.isDraggingSomething() && ((window as any).mode.isTablet() || (window as any).mode.isMobile()))
-			this.sendTransformCommand(point);
+		if (this.containerObject.isDraggingSomething()) {
+			if ((window as any).mode.isTablet() || (window as any).mode.isMobile())
+				this.sendTransformCommand(point);
+			else if ((window as any).mode.isDesktop() && (this.sectionProperties.closestX || this.sectionProperties.closestY)) {
+				// We need to snap to the guide-lines. So we send a positioning command after the mouse up event (desktop case).
+				// We don't need to do this on mobile because we always send the positioning commands there. No mouse events for mobile.
+				setTimeout(() => {
+					this.sendTransformCommand(point); // Send to back of the process queue so it performs after buttonup event is sent.
+				}, 10);
```

Comment:
in that case I prefer "0" so it is added to the queue immediately without risk of something between